src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RedirectedBasicWriter.cs

```diff
@@ -70,7 +72,39 @@ public override void WriteHtmlAttribute(CSharpRenderingContext context, HtmlAttr
 
         public override void WriteHtmlContent(CSharpRenderingContext context, HtmlContentIRNode node)
         {
-            _previous.WriteHtmlContent(context, node);
+            if (context.Options.DesignTimeMode)
+            {
+                _previous.WriteHtmlContent(context, node);
+                return;
+            }
+
+            const int MaxStringLiteralLength = 1024;
```

Comment:
All of these redirected bits are so ugly. I'm torn between leaving these as is vs. consolidating logic into extensions or helpers. If we plan to go down the path of not having redirect writers at all I'd be down to leave as is and nuke later; otherwise isolating would be better.